spec/models/manageiq/providers/redhat/infra_manager_spec.rb

```diff
@@ -78,4 +78,128 @@
       expect(described_class.extract_ems_ref_id("/ovirt-engine/api/vms/123")).to eq("123")
     end
   end
+
+  context "api versions" do
+    require 'ovirtsdk4'
+    let(:ems) { FactoryGirl.create(:ems_redhat_with_authentication) }
+    subject(:supported_api_versions) { ems.supported_api_versions }
+    context "#supported_api_versions" do
+      before(:each) do
+        Rails.cache.delete(ems.cache_key)
+        Rails.cache.write(ems.cache_key, cached_api_versions)
+      end
+
+      context "when no cached supported_api_versions" do
+        let(:cached_api_versions) { nil }
+        it 'calls the OvirtSDK4::Probe.probe' do
+          allow(OvirtSDK4::Probe).to receive(:probe).and_return([])
+          expect(OvirtSDK4::Probe).to receive(:probe)
+          supported_api_versions
+        end
+
+        it 'properly parses ProbeResults' do
+          allow(OvirtSDK4::Probe).to receive(:probe)
+            .and_return([OvirtSDK4::ProbeResult.new(:version => '3'),
+                         OvirtSDK4::ProbeResult.new(:version => '4')])
+          expect(supported_api_versions).to match_array(%w(3 4))
+        end
+      end
+
+      context "when cache of supported_api_versions is stale" do
+        let(:cached_api_versions) do
+          {
+            :created_at => Time.now.utc,
+            :value      => [3]
+          }
+        end
+
+        before(:each) do
+          allow(ems).to receive(:last_refresh_date)
+            .and_return(cached_api_versions[:created_at] + 1.day)
+        end
+
+        it 'calls the OvirtSDK4::Probe.probe' do
+          allow(OvirtSDK4::Probe).to receive(:probe).and_return([])
+          expect(OvirtSDK4::Probe).to receive(:probe)
+          supported_api_versions
+        end
+      end
+
+      context "when cache of supported_api_versions availabe" do
+        let(:cached_api_versions) do
+          {
+            :created_at => Time.now.utc,
+            :value      => [3]
+          }
+        end
+
+        before(:each) do
+          allow(ems).to receive(:last_refresh_date)
+            .and_return(cached_api_versions[:created_at] - 1.day)
+        end
+
+        it 'returns from cache' do
+          expect(supported_api_versions).to match_array([3])
+        end
+      end
+    end
+
+    context "#supports_the_api_version?" do
+      it "returns the supported api versions" do
+        allow(ems).to receive(:supported_api_versions).and_return([3])
+        expect(ems.supports_the_api_version?(3)).to eq(true)
+        expect(ems.supports_the_api_version?(6)).to eq(false)
+      end
+    end
+  end
+
+  context "supported features" do
+    let(:ems) { FactoryGirl.create(:ems_redhat) }
+    let(:supported_api_versions) { [3, 4] }
+    context "#process_api_features_support" do
+      before(:each) do
+        allow(SupportsFeatureMixin).to receive(:guard_queryable_feature).and_return(true)
```

Comment:
ah, ok, because you are testing against arbirtrary features.

See my comment above. I dont really like this generic feature handling. If you just add the two `supports` blocks for your two features you can get along without this whole spec context.